### PR TITLE
Quick and Dirty attempt at aligning object literals #365

### PR DESCRIPTION
**Pull request by clawconduce**
I should start by saying I'm not a fan of the code here, but I do like the output.

I want to get the ball rolling on #365 but I am definitely not familiar with the codebase, but my hope is that this will eventually lead to getting this issue resolved.

### CODE REVIEW

**Comment by clawconduce** [comment]
I'm not sure why AppVeyor failed.

I wanted to note that I just saw that Python changes are required as well.  I will port it over once the approach looks like it's reasonably close.

**Comment by clawconduce** [comment]
Is the code beautified?  What parameters are used to beautify the source?

**Comment by bitwiseman** [comment]
@clawconduce Thanks for contributing.  Disliking the code is perfectly acceptable and you are in good company.  It's a long road.

If you run, `./build` that will beautify the code, yes.   The settings are in `./jsbeautifyrc`.

File: js/lib/beautify.js [hunk 1]
```diff
@@ -1014,6 +1015,62 @@ function Beautifier(js_source_text, options) {
...
+                        }
+                        continue;
+                    }
+                    if (lookingAt === 'OPERATOR' && check_token.type == 'TK_OPERATOR') {
+                        lookingAt = 'VALUESTART';
+                        continue;
+                    }
+                    if (lookingAt === 'VALUESTART' || lookingAt === 'VALUE') {
+                        if (lookingAt === 'VALUESTART' && setKeyPadding !== null) {
+                            check_token.text = new Array(setKeyPadding - lastKeyLen + 1).join(' ') + check_token.text;
```
Comment (clawconduce) [inline js/lib/beautify.js]: this is my main dislike.  I tried setting whitespace_before to a string of whitespace, but that didn't appear to affect the output.  Is there a better way?

Comment (bitwiseman) [inline js/lib/beautify.js]: Whitespace_before is a value on the parsed token, but ignored during output except in the case of `keep_array_indentation = true`.  
see https://github.com/beautify-web/js-beautify/blob/master/js/src/javascript/beautifier.js#L413

**Comment by bitwiseman** [comment]
@clawconduce - When do you think you'll have a chance to work on this?

**Comment by clawconduce** [comment]
I probably won't get the chance for a few weeks at least.  If someone else wants to pick this up, I have no problems with that too, though I will hopefully get some time sooner rather than later.

I have been thinking in the back of my mind how to limit when this happens, I realized that when the values are functions, especially long functions, this sort of indentation is not desirable, so either objects with functions get skipped completely, or maybe there's a parameter that will disable this if any value in an object is a function more than N lines long.

**Comment by bitwiseman** [comment]
@clawconduce - Any updates?

**Comment by clawconduce** [comment]
Sorry for the long time between updates.

It seems my todo list is growing quicker than I am pulling stuff off.  I don't myself getting this merge ready anytime soon.

I'll close this PR - hopefully someone sees #396 and can use this as a starting point.

**Pull request closed without merging**